base/compiler/inferencestate.jl

```diff
@@ -192,6 +192,8 @@ function sptypes_from_meth_instance(linfo::MethodInstance)
                     ty = UnionAll(tv, Type{tv})
                 end
             end
+        elseif isa(v, Core.VarargMarker)
+            ty = Int
```

Comment:
Fixes #37316?

Reply:
Yes

Reply:
It doesn't look like this was enough:
```julia
julia> f(::NTuple{N}) where {N} = N
f (generic function with 1 method)

julia> code_warntype(f, Tuple{Tuple})
Variables
  #self#::Core.Const(f)
  #unused#::Tuple{Vararg{T, N}} where T where N

Body::Any
1 ─     return $(Expr(:static_parameter, 1))
```
Or is this check flawed in some way?